ansible/modules/hashivault/hashivault_auth_method.py

```diff
@@ -144,7 +144,7 @@ def hashivault_auth_method(module):
             config['force_no_cache'] = False
         if 'token_type' not in config:
             config['token_type'] = 'default-service'
-        if current_state['data'] != config:
+        if not current_state['data'].items() >= config.items():
```

Comment:
I don't get why you are using >= to here.

Reply:
I'm checking if the state that Vault returns has the same items as your module data or even more items. If your module items has different items than the Vault API, then something will be changed.

The issue was that Vault adds default fields or has fields that you don't want/need to change, so this code only checks that the fields that you want to change are the same as from the Vault API.